lib/board.rb

```diff
@@ -0,0 +1,43 @@
+class Board
+  attr_accessor :cells
+  def initialize
+    self.reset!
+  end
+
+  def reset!
+    self.cells=Array.new(9, " ")
+  end
+
+  def display
+    board=self.cells
+      puts " #{board[0]} | #{board[1]} | #{board[2]} "
+      puts "-----------"
+      puts " #{board[3]} | #{board[4]} | #{board[5]} "
+      puts "-----------"
+      puts " #{board[6]} | #{board[7]} | #{board[8]} "
+    end
+
+  def position(n)
+    self.cells[n.to_i-1]
+  end
+
+  def full?
+    !self.cells.include?(" ")
+  end
+
+  def turn_count
```

Comment:
Consider using the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```